media/js/base/mozilla-fxa-link.js

```diff
@@ -34,21 +41,68 @@ if (typeof window.Mozilla === 'undefined') {
         return href;
     };
 
+    /**
+     * Returns the hostname for a given URL.
+     * @param {String} url.
+     * @returns {String} hostname.
+     */
+    FxaLink.getHostName = function(url) {
+        var matches = url.match(/^https?:\/\/(?:[^/?#]+)(?:[/?#]|$)/i);
+        return matches && matches[0];
+    };
+
+    /**
+     * Intercept event handler for FxA links and forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaLink.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var search = new URL(event.target.href).search;
+        var urlParams = new window._SearchParams(search).params;
+        var extraURLParams = {};
+        var allowedChars = /^[\w/.%-]+$/;
+        // There are no `flowId` params below because only the `fxa-form` starts a new flow.
```

Comment:
We still need to handle `flowId` params on links, as they can be added to main call-to-action links via `mozilla-fxa-product-button.js`.